### ISSUE TITLE
IndexError: list index out of range

### ISSUE THREAD

**Issue by ashokpatowary** [ablab/IsoQuant#86]
Thus was reported earlier also and I believe was fixed in prev version; but I am still getting it
version used IsoQuant 3.2.0
```
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/u/local/apps/python/3.9.6/gcc-4.8.5/lib/python3.9/concurrent/futures/process.py", line 243, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/u/local/apps/python/3.9.6/gcc-4.8.5/lib/python3.9/concurrent/futures/process.py", line 202, in _process_chunk
    return [fn(*args) for args in chunk]
  File "/u/local/apps/python/3.9.6/gcc-4.8.5/lib/python3.9/concurrent/futures/process.py", line 202, in <listcomp>
    return [fn(*args) for args in chunk]
  File "/u/project/geschwind/ashokpat/Tools/IsoQuant/src/dataset_processor.py", line 112, in collect_reads_in_parallel
    for gene_info, assignment_storage in alignment_collector.process():
  File "/u/project/geschwind/ashokpat/Tools/IsoQuant/src/alignment_processor.py", line 242, in process
    for res in self.forward_alignments(alignment_storage):
  File "/u/project/geschwind/ashokpat/Tools/IsoQuant/src/alignment_processor.py", line 261, in forward_alignments
    yield self.process_alignments_in_region(new_region, alignments)
  File "/u/project/geschwind/ashokpat/Tools/IsoQuant/src/alignment_processor.py", line 269, in process_alignments_in_region
    assignment_storage = self.process_genic(alignment_storage, gene_info)
  File "/u/project/geschwind/ashokpat/Tools/IsoQuant/src/alignment_processor.py", line 345, in process_genic
    read_assignment = assigner.assign_to_isoform(read_id, alignment_info.combined_profile)
  File "/u/project/geschwind/ashokpat/Tools/IsoQuant/src/long_read_assigner.py", line 443, in assign_to_isoform
    assignment = self.match_inconsistent(read_id, combined_read_profile)
  File "/u/project/geschwind/ashokpat/Tools/IsoQuant/src/long_read_assigner.py", line 593, in match_inconsistent
    isoform_matches = self.create_consistent_matches(read_matches, best_isoforms, combined_read_profile)
  File "/u/project/geschwind/ashokpat/Tools/IsoQuant/src/long_read_assigner.py", line 631, in create_consistent_matches
    isoform_match = self.categorize_correct_splice_match(combined_read_profile, isoform_id)
  File "/u/project/geschwind/ashokpat/Tools/IsoQuant/src/long_read_assigner.py", line 353, in categorize_correct_splice_match
    elif self.is_fsm(read_region, isoform_id):
  File "/u/project/geschwind/ashokpat/Tools/IsoQuant/src/long_read_assigner.py", line 341, in is_fsm
    return contains(read_region, (isoform_introns[0][0], isoform_introns[-1][1]))
IndexError: list index out of range
"""
The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/u/home/a/ashokpat/project-geschwind/Tools/IsoQuant/isoquant.py", line 687, in <module>
    main(sys.argv[1:])
  File "/u/home/a/ashokpat/project-geschwind/Tools/IsoQuant/isoquant.py", line 681, in main
    run_pipeline(args)
  File "/u/home/a/ashokpat/project-geschwind/Tools/IsoQuant/isoquant.py", line 637, in run_pipeline
    dataset_processor.process_all_samples(args.input_data)
  File "/u/project/geschwind/ashokpat/Tools/IsoQuant/src/dataset_processor.py", line 364, in process_all_samples
    self.process_sample(sample)
  File "/u/project/geschwind/ashokpat/Tools/IsoQuant/src/dataset_processor.py", line 387, in process_sample
    self.collect_reads(sample)
  File "/u/project/geschwind/ashokpat/Tools/IsoQuant/src/dataset_processor.py", line 444, in collect_reads
    for storage, read_groups in results:
  File "/u/local/apps/python/3.9.6/gcc-4.8.5/lib/python3.9/concurrent/futures/process.py", line 559, in _chain_from_iterable_of_lists
    for element in iterable:
  File "/u/local/apps/python/3.9.6/gcc-4.8.5/lib/python3.9/concurrent/futures/_base.py", line 608, in result_iterator
    yield fs.pop().result()
  File "/u/local/apps/python/3.9.6/gcc-4.8.5/lib/python3.9/concurrent/futures/_base.py", line 445, in result
    return self.__get_result()
  File "/u/local/apps/python/3.9.6/gcc-4.8.5/lib/python3.9/concurrent/futures/_base.py", line 390, in __get_result
    raise self._exception
IndexError: list index out of range
```

**Comment by andrewprzh** [comment]
Dear @ashokpatowary 

Seems to be the same exception, but different issue. Is it possible for you to to send the entire isoquant.log and the GTF file you are using?

Best
Andrey

**Comment by ashokpatowary** [comment]
Thanks @andrewprzh 

Can you share ur email please?

**Comment by andrewprzh** [comment]
Yep, it should be visible in my GitHub profile.

**Comment by ashokpatowary** [comment]
I have send the files.

Thanks

**Comment by andrewprzh** [comment]
Thanks for the data!

The annotation contains multiple cases when adjacent exons have an intron of length 0 bp between them (315 cases in 424 transcripts). In contrast, standard human gencode has no such cases. Although it does not sound from biological point of view, it is no reason for a tool to fail :)

I'll try to figure out how to fix this.

Best
Andrey

**Comment by andrewprzh** [comment]
Should be fixed in IsoQuant 3.3.

Best
Andrey
